lib/Core/Memory.h

```diff
@@ -174,14 +174,17 @@ class ObjectState {
 
   uint8_t *concreteStore;
 
-  // XXX cleanup name of flushMask (its backwards or something)
+  // concreteMask[byte] is set if byte is known to be concrete
   BitArray *concreteMask;
 
-  // mutable because may need flushed during read of const
-  mutable BitArray *flushMask;
-
+  // knownSymbolics[byte] holds the symbolic expression for byte,
+  // if byte is known to be symbolic
   ref<Expr> *knownSymbolics;
 
+  // unflushedMask[byte] is set if byte is unflushed
+  // mutable because may need flushed during read of const
+  mutable BitArray *unflushedMask;
```

Comment:
By using `std::unique_ptr<BitArray>` the lines from the constructors/destructors (besides OS(&os)) could just be removed.

Reply:
I think this should be done in a separate commit.

Reply:
That's kind of a Catch-22 when it is discouraged to modernise code just for the sake of it. :wink: